### ISSUE TITLE
Breakpoints no longer working

### ISSUE THREAD

**Issue by munkyonline** [xdebug/vscode-php-debug#678]
PHP version: 7.4.11
Xdebug version: 3.1.0
VS Code extension version: 1.21.0

**Your launch.json:**
`{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for XDebug",
            "type": "php",
            "request": "launch",
            "port": 9000,
            "stopOnEntry": false,
            "pathMappings": {
                "/var/www/vhosts/domain/httpdocs": "${workspaceRoot}/"
            },
            "log": true
        },
        {
            "name": "Launch currently open script",
            "type": "php",
            "request": "launch",
            "program": "${file}",
            "cwd": "${fileDirname}",
            "port": 9000,
        },
    ],
}`

**Xdebug php.ini config:**
xdebug.mode = debug,trace,develop
xdebug.start_with_request = yes
xdebug.client_port = 9000

Xdebug logfile (from setting `xdebug.log` in php.ini): Setting this doesn't create a log for me.
VS Code extension logfile (from setting `"log": true` in launch.json):
[debugconsole.txt](https://github.com/xdebug/vscode-php-debug/files/7384855/debugconsole.txt)

Code snippet to reproduce:

```php
$i = [1,2,3,4,5];
var_dump($i);
phpinfo(); 
```

Hi, I'm not entirely sure what's happened, it's probably something simple that I've overlooked, but breakpoints are no longer working at all. I've tried extension versions going back to 1.16.1. This is the same on both my work Mac, and my Windows 11 laptop that both use the same cloned VM. Also occurs in different workspaces.

The Mac is using vscode: 1.47.3, and a slightly older Xdebug version, but otherwise the settings and setup are the same. The launch.json has been working for months, so I don't think its a path mapping issue.

I have Xdebug running in a virtualbox environment, and I use a SSH tunnel to connect, without this I can see in the error log:
`Xdebug: [Step Debug] Could not connect to debugging client. Tried: localhost:9000 (through xdebug.client_host/xdebug.client_port) :-(`

Also the var_dump is showing the extra data, so I know Xdebug is working, and it's showing the Development Helpers, Step Debugger and Tracing are enabled in phpinfo.

**Comment by zobo** [comment]
Hi!
I can see from `debugconsole.txt` that the plugin starts listening on port 9000. No connections though. The Xdebug log says it cant connect to `localhost:9000`.
You are mentioning a VM..
Are you using VSCode Remote? Do you use some other form of remote file access? Are the running PHP and VS Code on different hosts/VMs?
Are you running the PHP code via a web server or command line?

Please describe your setup in regards of where VS Code runs, where the PHP code resides and where the PHP process runs and how the network is set up between these components.

Thanks!

**Comment by munkyonline** [comment]
Thanks for getting back to me zobo. 

My setup:
I'm running VScode on Windows 11 which is the host and the VM guest is CentOS 7. They are connected via a bridged connection. The project files are stored in Windows and are shared with the guest. The guest is running the LAMP stack with Xdebug. The setup is the same on my Mac.

To 'forward' the guest 9000 port to the host port 9000 I SSH into the guest using:
`ssh -R 9000:9000 user@guest`

That localhost:9000 error is from the apache/php log, the error stops once i've set up the SSH tunnel. This was working fine before.

The only changes that have occurred recently are upgrading from Win 10, and VSCode updating. No major updates on the Mac that I know of.

This connects in Windows host, and stopping the debugger ends the connection:
`telnet localhost 9000`

I've added rules in the Windows firewall so that port 9000 is definitely open.

On the guest 
```
sudo firewall-cmd --list-ports
443/tcp 9000/tcp
```
On the guest running netstat -tulnp with the ssh tunnel enabled:
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:9000          0.0.0.0:*               LISTEN      -

```
Then on guest this connects
`telnet localhost 9000`

I've also tried adding xdebug_break() but that doesn't work either.
